src/System/Validation/ValidatedNotNullAttribute.cs

```diff
@@ -0,0 +1,13 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+
+namespace Microsoft.Omex.System.Validation
+{
+	/// <summary>
+	/// Indicates to Code analysis tools that a method validates a particular parameter to not be null
+	/// </summary>
+	[AttributeUsage(AttributeTargets.Parameter, AllowMultiple = false, Inherited = false)]
+	internal sealed class ValidatedNotNullAttribute : Attribute { }

```

Comment:
>ValidatedNotNullAttribute [](start = 23, length = 25)

I understand the rationale for this, but the naming doesn't seem quite right. The elements marked with this are potentially null and you're validating that they're not null, whereas the naming here indicates that it has been validated as not being null. Can you use an alternative name?

Reply:
The naming is unfortunately nothing I can change. That is what the analysers look at; whether an attribute with the name `ValidatedNotNull` is present on the parameter.

Reply:
It might be worth briefly mentioning this in the comment as well.